src/utility/Assert.h

```diff
@@ -0,0 +1,38 @@
+/* Copyright 2019 Ole Kliemann, Malte Kliemann
+ *
+ * This file is part of DrMock.
+ *
+ * DrMock is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * DrMock is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with DrMock.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#ifndef DRMOCK_SRC_UTILITY_ASSERT_H
+#define DRMOCK_SRC_UTILITY_ASSERT_H
+
+#define DR_ASSERT(expr) do { \
+  if (not expr) \
+  { \
+    std::string msg{}; \
+    msg = "assert: "; \
+    msg += __FILE__; \
+    msg += ": "; \
+    msg += std::to_string(__LINE__); \
+    msg += ": "; \
+    msg += __func__; \
+    msg += ": "; \
+    msg += #expr; \
+    throw std::logic_error{msg}; \
+  } \
+} while (false)
```

Comment:
`throw` instead of `std::abort`: I want to be able to still handle it. Also I want it to stay in production code. But it's probably too specific, so I removed it for now.

Regarding the `AssertionFailure`: It's a good idea, but should it be derived from `std::exception` or would that cause the same issue?

Reply:
If it stays in production code, however, I'd say the name is "wrong". Anything called assert I would expect to be a debug utility.

Deriving `AssertionFailure` from `std::exception` is fine, I think. The only way I see you run into trouble here is if someone does `try { /* ... */ } catch (std::exception) { /* ... */ }`, which is a bad idea anyways.